patchwork-extensions/src/main/java/net/minecraftforge/common/extensions/IForgeItem.java

```diff
@@ -645,41 +682,11 @@ default boolean canContinueUsing(ItemStack oldStack, ItemStack newStack) {
 	 */
 	@Nullable
 	default String getCreatorModId(ItemStack itemStack) {
-		Item item = itemStack.getItem();
-		Identifier registryName = Registry.ITEM.getId(item);
-		String modId = registryName == null ? null : registryName.getNamespace();
-
-		if ("minecraft".equals(modId)) {
-			if (item instanceof EnchantedBookItem) {
-				ListTag enchantmentsNbt = EnchantedBookItem.getEnchantmentTag(itemStack);
-
-				if (enchantmentsNbt.size() == 1) {
-					CompoundTag nbttagcompound = enchantmentsNbt.getCompoundTag(0);
-					Identifier resourceLocation = Identifier.tryParse(nbttagcompound.getString("id"));
-
-					if (resourceLocation != null && Registry.ENCHANTMENT.containsId(resourceLocation)) {
-						return resourceLocation.getNamespace();
-					}
-				}
-			} else if (item instanceof PotionItem || item instanceof TippedArrowItem) {
-				Potion potionType = PotionUtil.getPotion(itemStack);
-				Identifier resourceLocation = Registry.POTION.getId(potionType);
-
-				if (resourceLocation != null) {
-					return resourceLocation.getNamespace();
-				}
-			} else if (item instanceof SpawnEggItem) {
-				Identifier resourceLocation = Registry.ENTITY_TYPE.getId(((SpawnEggItem) item).getEntityType(null));
-
-				if (resourceLocation != null) {
-					return resourceLocation.getNamespace();
-				}
-			}
-		}
-
-		return modId;
+		final Identifier id = Registry.ITEM.getId(itemStack.getItem());
+		return !itemStack.isEmpty() && Registry.ITEM.getDefaultId().equals(id) ? null : id.getNamespace();
```

Comment:
Shouldn't this be an `||` because the item not existing *or* the stack being empty would be conditions for returning null?
```suggestion
		return itemStack.isEmpty() || Registry.ITEM.getDefaultId().equals(id) ? null : id.getNamespace();
```